src/api/serverOperations/selectProductInfo.ts

```diff
@@ -0,0 +1,42 @@
+import { ProductInfo } from "@/productInfos";
+
+function compareByCriteria(info1: ProductInfo, info2: ProductInfo, criteriaName: string): number {
+  switch (criteriaName) {
+    case "date": {
+      if (info1.additionDate < info2.additionDate) return 1;
+      if (info1.additionDate > info2.additionDate) return -1;
+      return 0;
+    }
+    case "price":
+      return info1.price - info2.price;
+    case "mark":
+      return info1.mark - info2.mark;
+    default: {
+      if (info1.name > info2.name) return 1;
+      if (info1.name < info2.name) return -1;
+      return 0;
+    }
+  }
+}
+
+function select(field: string, amount: number, arr: Array<ProductInfo>) {
+  console.log(`${field}: ${amount}`);
+  const sortedArr = arr.sort((prInfo1, prInfo2) => compareByCriteria(prInfo1, prInfo2, field));
+  return sortedArr.slice(0, amount); /* .concat([
+    {
+      name: field,
+      price: amount,
+      mark: 4,
+      image: "images/Cyberpunk.jpg",
+      description: "Best game",
+      ageCategory: 0,
+      isPC: true,
+      isXBox: false,
+      isPS: false,
+      additionDate: new Date("2019-02-02"),
+      key: 1,
+    },
+  ]);*/
+}
```

Comment:
what is it?

Reply:
it I used to return server-processed request info, because it was not shown in server console. Now it is removed